federation/pkg/dnsprovider/providers/aws/route53/rrsets.go

```diff
@@ -0,0 +1,176 @@
+	"k8s.io/kubernetes/federation/pkg/dnsprovider/rrstype"
+)
+
+type ResourceRecordSets struct {
+	zone *Zone
+}
+
+func (rrsets ResourceRecordSets) List() ([]dnsprovider.ResourceRecordSet, error) {
+	input := route53.ListResourceRecordSetsInput{
+		HostedZoneId: rrsets.zone.impl.Id,
+	}
+	response, err := rrsets.zone.zones.interface_.service.ListResourceRecordSets(&input)
+	// TODO: Handle truncated responses
+	if err != nil {
+		return nil, err
+	}
+	list := make([]dnsprovider.ResourceRecordSet, len(response.ResourceRecordSets))
+	for i, rrset := range response.ResourceRecordSets {
+		list[i] = &ResourceRecordSet{rrset, &rrsets}
+	}
+	return list, nil
+}
+
+func (rrsets ResourceRecordSets) Add(rrset dnsprovider.ResourceRecordSet) (dnsprovider.ResourceRecordSet, error) {
+	service := rrsets.zone.zones.interface_.service
+	input := &route53.ChangeResourceRecordSetsInput{
+		ChangeBatch: &route53.ChangeBatch{ // Required
+			Changes: []*route53.Change{ // Required
+				{ // Required
+					Action: aws.String("CREATE"), // Required
+					ResourceRecordSet: &route53.ResourceRecordSet{ // Required
+						Name: aws.String(rrset.Name()),         // Required
+						Type: aws.String(string(rrset.Type())), // Required
+						/*  Unsupported fields commented out for now, to make extension easier later (and hopefully inspire other to :-).
+						                                                    Can be removed if deemed too messy here.
+													AliasTarget: &route53.AliasTarget{
+														DNSName:              aws.String("DNSName"),    // Required
+														EvaluateTargetHealth: aws.Bool(true),           // Required
+														HostedZoneId:         aws.String("ResourceId"), // Required
+													},
+													Failover: aws.String("ResourceRecordSetFailover"),
+													GeoLocation: &route53.GeoLocation{
+														ContinentCode:   aws.String("GeoLocationContinentCode"),
+														CountryCode:     aws.String("GeoLocationCountryCode"),
+														SubdivisionCode: aws.String("GeoLocationSubdivisionCode"),
+													},
+													HealthCheckId: aws.String("HealthCheckId"),
+													Region:        aws.String("ResourceRecordSetRegion"),
+						*/
+						ResourceRecords: []*route53.ResourceRecord{
+							{ // Required
+								Value: aws.String(rrset.Rrdatas()[0]), // Required
+							},
+							// More values...
+						},
+						/*
+							SetIdentifier: aws.String("ResourceRecordSetIdentifier"),
+
+						*/
+						TTL: aws.Int64(rrset.Ttl()),
+						/*
+							TrafficPolicyInstanceId: aws.String("TrafficPolicyInstanceId"),
+							Weight:                  aws.Int64(1),
+						*/
+					},
+				},
+				// More values...
+			},
+			Comment: aws.String("Ubernetes federated service"),
+		},
+		HostedZoneId: aws.String(*rrset.(ResourceRecordSet).rrsets.zone.impl.Id), // Required
+	}
+	_, err := service.ChangeResourceRecordSets(input)
+	if err != nil {
+		// Cast err to awserr.Error to get the Code and
```

Comment:
No, it's more a note to future developers who come after me, who might want to extract error codes.  Can delete the comment if you really prefer.